File: backend/app/routes/analytics.py
"""Analytics blueprint — read-only endpoints."""
import datetime

from flask import Blueprint, jsonify, request

from app.services.analytics_service import AnalyticsService

analytics_bp = Blueprint("analytics", __name__)


def _current_month() -> str:
    return datetime.date.today().strftime("%Y-%m")


@analytics_bp.route("/summary", methods=["GET"])
def summary():
    month = request.args.get("month", _current_month())
    return jsonify(AnalyticsService.monthly_summary(month)), 200


@analytics_bp.route("/breakdown", methods=["GET"])
def breakdown():
    month = request.args.get("month", _current_month())
    return jsonify(AnalyticsService.category_breakdown(month)), 200


@analytics_bp.route("/trends", methods=["GET"])
def trends():
    months = request.args.get("months", 6, type=int)
    return jsonify(AnalyticsService.monthly_trends(months=min(months, 24))), 200

File: frontend/src/api/analytics.js
import client from "./client";

export const getSummary = (month) =>
    client.get("/analytics/summary", { params: { month } });

export const getBreakdown = (month) =>
    client.get("/analytics/breakdown", { params: { month } });

export const getTrends = (months = 6) =>
    client.get("/analytics/trends", { params: { months } });

export const categorizeDescription = (description) =>
    client.post("/ai/categorize", { description });
